osu.Game.Tournament/Screens/SetupScreen.cs

```diff
@@ -0,0 +1,142 @@
+using osu.Game.Overlays;
+using osu.Game.Screens.Edit.Setup.Components.LabelledComponents;
+using osu.Game.Tournament.IPC;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Tournament.Screens
+{
+    public class SetupScreen : TournamentScreen, IProvideVideo
+    {
+        private FillFlowContainer fillFlow;
+
+        private LoginOverlay loginOverlay;
+
+        [Resolved]
+        private MatchIPCInfo ipc { get; set; }
+
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            InternalChild = fillFlow = new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Direction = FillDirection.Vertical,
+                Padding = new MarginPadding(10),
+                Spacing = new Vector2(10),
+            };
+
+            api.LocalUser.BindValueChanged(_ => Schedule(reload));
+            reload();
+        }
+
+        private void reload()
+        {
+            var fileBasedIpc = ipc as FileBasedIPC;
+
+            fillFlow.Children = new Drawable[]
+            {
+                new ActionableInfo
+                {
+                    Label = "Current IPC source",
+                    ButtonText = "Refresh",
+                    Action = () =>
+                    {
+                        fileBasedIpc?.LocateStableStorage();
+                        reload();
+                    },
+                    Value = fileBasedIpc?.Storage?.GetFullPath(string.Empty) ?? "Not found",
+                    Failing = fileBasedIpc?.Storage == null,
+                    Description = "The osu!stable installation which is currently being used as a data source. If a source is not found, make sure you have created an empty ipc.txt in your stable cutting-edge installation, and that it is registered as the default osu! install."
+                },
+                new ActionableInfo
+                {
+                    Label = "Current User",
+                    ButtonText = "Change Login",
+                    Action = () =>
+                    {
+                        api.Logout();
+
+                        if (loginOverlay == null)
+                        {
+                            AddInternal(loginOverlay = new LoginOverlay
+                            {
+                                Anchor = Anchor.TopRight,
+                                Origin = Anchor.TopRight,
+                            });
+                        }
+
+                        loginOverlay.State.Value = Visibility.Visible;
+                    },
+                    Value = api?.LocalUser.Value.Username,
+                    Failing = api?.IsLoggedIn != true,
+                    Description = "In order to access the API and display metadata, a login is required."
+                }
+            };
+        }
+
+        private class ActionableInfo : LabelledComponent
```

Comment:
What do you think about renaming this to `LabelledFailableAction` or similar?

Reply:
It's very local, it's fine.